google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurator.java

```diff
@@ -115,6 +155,24 @@ public AppEngineDeploymentConfiguration createDefaultConfiguration(
     return new AppEngineDeploymentRunConfigurationEditor(
         project,
         source,
-        new CloudSdkAppEngineHelper(new File(server.getConfiguration().getCloudSdkHomePath())));
+        new CloudSdkAppEngineHelper(
+            project,
+            new File(server.getConfiguration().getCloudSdkHomePath()),
+            getEnvironment())
+    );
+  }
+
+  /**
+   * Lazily initialize the environment because this class is instantiated on IDE boot time
+   * and the project may not have yet been fully constructed.
+   */
+  public Environment getEnvironment() {
```

Comment:
Sorry I missed this - yes facets are at the module level. I'm thinking this Environment concept should eventually live on the AE facet once we merge in the plugins. Currently we don't have an explicitly defined facet so I'm thinking we can leave this as is, with a todo to refactor this once merged. Are you ok with this approach?